dotnet/samples/Concepts/Filtering/AzureOpenAI_DeploymentSwitch.cs

```diff
@@ -0,0 +1,110 @@
+        var handler = new LoggingHandler(new HttpClientHandler(), this.Output);
+        var httpClient = new HttpClient(handler);
+
+        // Create KernelBuilder with an auto function invocation filter
+        var kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.Services.AddSingleton<IAutoFunctionInvocationFilter>(new AutoFunctionInvocationFilter(this.Output));
+
+        // Define the endpoints for the two Azure OpenAI services
+        var endpoint1 = "https://contoso-openai-eastus.openai.azure.com/";
+        var endpoint2 = "https://contoso-openai-swedencentral.openai.azure.com/";
+
+        // Add Azure OpenAI chat completion services
+        kernelBuilder.AddAzureOpenAIChatCompletion(
+            serviceId: "eastus",
+            deploymentName: "gpt-4o-mini",
+            endpoint: endpoint1,
+            credentials: new DefaultAzureCredential(),
+            httpClient: httpClient,
+            modelId: TestConfiguration.AzureOpenAI.ChatModelId);
+        kernelBuilder.AddAzureOpenAIChatCompletion(
+            serviceId: "swedencentral",
+            deploymentName: "gpt-4o",
+            endpoint: endpoint2,
+            credentials: new DefaultAzureCredential(),
+            httpClient: httpClient,
+            modelId: TestConfiguration.AzureOpenAI.ChatModelId);
+        var kernel = kernelBuilder.Build();
+
+        kernel.ImportPluginFromFunctions("HelperFunctions",
+        [
+            kernel.CreateFunctionFromMethod(() => "Brown", "GetEyeColor", "Retrieves eye color for the current user."),
+            kernel.CreateFunctionFromMethod(() => DateTime.UtcNow.ToString("R"), "GetCurrentDateTimeInUtc", "Retrieves the current date time in UTC."),
+        ]);
+
+        OpenAIPromptExecutionSettings settings = new()
+        {
+            ServiceId = "swedencentral",
+            FunctionChoiceBehavior = FunctionChoiceBehavior.Auto()
+        };
+        kernel.Data.Add("service_id", "swedencentral");
+        var reply = await kernel.InvokePromptAsync("What time is it and what is my eye color and what time is it?", new(settings));
+
+        Console.WriteLine(reply);
+    }
+
+    private sealed class AutoFunctionInvocationFilter(ITestOutputHelper output) : IAutoFunctionInvocationFilter
+    {
+        public async Task OnAutoFunctionInvocationAsync(AutoFunctionInvocationContext context, Func<AutoFunctionInvocationContext, Task> next)
+        {
+            var kernel = context.Kernel;
+            var chatHistory = context.ChatHistory;
+            var functionCalls = FunctionCallContent.GetFunctionCalls(context.ChatHistory.Last());
+
+            if (kernel.Data.TryGetValue("service_id", out object? serviceId) && serviceId is not null && "swedencentral".Equals(serviceId.ToString(), StringComparison.Ordinal))
+            {
+                bool includesGetEyeColor = functionCalls.Any(fc => fc.FunctionName.Equals("GetEyeColor", StringComparison.Ordinal));
+                if (!includesGetEyeColor)
```

Comment:
I think some comments will be useful here for better understanding:
```suggestion
                
                // For the "GetEyeColor" function, switch to a different deployment. 
                // If the function is not present in the collection of function calls, proceed with the request as usual.
                if (!includesGetEyeColor)
```